### PR TITLE
fix: map username/label for federated users (AR-2036)

### PR DESCRIPTION
**Pull request by yamilmedina**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2036" title="AR-2036" target="_blank">BugAR-2036</a>  No domain is displayed in search result for users from another backend
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

This PR enables showing information about the domain for federated users

### Attachments (Optional)

Screenshot_1658848129

Screenshot_1658848125

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-799/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3988175)(3988175)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3988175)(3988175)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-799/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3988338)(3988338)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3988338)(3988338)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/userprofile/common/UsernameMapper.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.wire.android.ui.userprofile.common
+
+import com.wire.kalium.logic.data.user.OtherUser
+import com.wire.kalium.logic.data.user.type.UserType
+
+object UsernameMapper {
+
+    fun mapUserLabel(otherUser: OtherUser): String = with(otherUser) {
```
Comment (Garzas) [inline app/src/main/kotlin/com/wire/android/ui/userprofile/common/UsernameMapper.kt:8]: Maybe we can write some test for that mapper?

**Review by Garzas — APPROVED** [review]
LGTM 🎸 
Just one comment about 🧪

**Pull request merged**